packages/backend/src/ee/services/AiAgentService.ts

```diff
@@ -1642,7 +1654,70 @@ export class AiAgentService {
             await this.aiAgentModel.createToolResults(data);
         };
 
+        const findDashboards: FindDashboardsFn = async (args) => {
+            const searchResults = await this.searchModel.searchDashboards(
+                projectUuid,
+                args.dashboardSearchQuery.label,
+            );
+
+            // Apply permission filtering - check space access for each dashboard
+            const spaceUuids = [
+                ...new Set(
+                    searchResults.map((dashboard) => dashboard.spaceUuid),
+                ),
+            ];
+
+            const spaces = await Promise.all(
+                spaceUuids.map((spaceUuid) =>
+                    this.spaceModel.getSpaceSummary(spaceUuid),
+                ),
+            );
+
+            const spacesAccess = await this.spaceModel.getUserSpacesAccess(
+                user.userUuid,
+                spaces.map((s) => s.uuid),
+            );
+
+            const filterDashboard = (dashboard: typeof searchResults[0]) => {
+                const itemSpace = spaces.find(
+                    (s) => s.uuid === dashboard.spaceUuid,
+                );
+                return (
+                    itemSpace &&
+                    hasViewAccessToSpace(
+                        user,
+                        itemSpace,
+                        spacesAccess[dashboard.spaceUuid] ?? [],
+                    )
+                );
+            };
+
+            const filteredResults = searchResults.filter(filterDashboard);
+
+            const totalResults = filteredResults.length;
+            const totalPageCount = Math.ceil(totalResults / args.pageSize);
+
+            // Apply pagination
+            const startIndex = (args.page - 1) * args.pageSize;
+            const endIndex = startIndex + args.pageSize;
+            const paginatedResults = filteredResults.slice(
+                startIndex,
+                endIndex,
+            );
```

Comment:
I think you can skip this, results will always be limited to 10:

https://github.com/lightdash/lightdash/blob/9f98c0eafc2668db06b9987ced6e748ae2c599b4/packages/backend/src/models/SearchModel/index.ts#L165-L169

But let's keep the interface (return value)